src/Sentry/Internal/Extensions/JsonExtensions.cs

```diff
@@ -415,13 +463,46 @@ public static void Deconstruct(this JsonProperty jsonProperty, out string name,
         {
             writer.WriteStringValue(dto);
         }
+        else if (value is TimeSpan timeSpan)
+        {
+            writer.WriteStringValue(timeSpan.ToString("g", CultureInfo.InvariantCulture));
+        }
+#if NET6_0_OR_GREATER
+        else if (value is DateOnly date)
+        {
+            writer.WriteStringValue(date.ToString("O", CultureInfo.InvariantCulture));
+        }
+        else if (value is TimeOnly time)
+        {
+            writer.WriteStringValue(time.ToString("HH:mm:ss.FFFFFFF", CultureInfo.InvariantCulture));
+        }
+#endif
         else if (value is IFormattable formattable)
         {
             writer.WriteStringValue(formattable.ToString(null, CultureInfo.InvariantCulture));
         }
         else
         {
-            JsonSerializer.Serialize(writer, value, SerializerOptions);
+            if (!JsonPreserveReferences)
```

Comment:
```suggestion
            if (JsonPreserveReferences is null)
```

The original is 100% correct. I find reading `something is null` makes it idiot-proof (like me 😀).

Reply:
This one is just a regular boolean, not a nullable boolean.